core/src/main/scala/fs2/StreamCore.scala

```diff
@@ -471,9 +471,9 @@ object StreamCore {
 
   private[fs2]
   def runCleanup[F[_]](cleanups: Iterable[Free[F,Either[Throwable,Unit]]]): Free[F,Either[Throwable,Unit]] = {
-    // note - run cleanup actions in LIFO order, later actions run first
+    // note - run cleanup actions in FIFO order, but avoid left-nesting binds
     // all actions are run but only first error is reported
```

Comment:
ok for now, but perhaps we can build and exception that will collect all throwables laters?